src/components/IconButton/IconButton.tsx

```diff
@@ -0,0 +1,82 @@
+import * as React from 'react'
+import propConnect from '../PropProvider/propConnect'
+import Button, { Props as ButtonPropsInterface } from '../Button/ButtonV2'
+import Icon from '../Icon'
+import { classNames } from '../../utilities/classNames'
+import { noop } from '../../utilities/other'
+import { IconButtonUI } from './IconButton.css'
+import { COMPONENT_KEY } from './IconButton.utils'
+
+export interface Props extends ButtonPropsInterface {
+  icon: string
+  iconSize: number
+  isBorderless: boolean
+  innerRef: (node: HTMLElement) => void
+  withCaret: boolean
+}
+
+export class IconButton extends React.PureComponent<Props> {
+  static className = 'c-IconButton'
+  static defaultProps = {
+    ...Button.defaultProps,
+    icon: 'search',
+    iconSize: 24,
+    innerRef: noop,
+    isBorderless: true,
+    kind: 'default',
+    size: 'md',
+    shape: 'circle',
+    withCaret: false,
+  }
+
+  getClassName() {
+    const { className, kind, icon, isBorderless } = this.props
+
+    return classNames(
+      IconButton.className,
+      icon && `is-icon-${icon}`,
+      kind && `is-kind-${kind}`,
+      isBorderless && 'is-borderless',
+      className
+    )
+  }
+
+  getIconSize(): number {
+    const { iconSize, size } = this.props
+
+    if (size === 'xs') {
+      return 16
+    }
+    if (size === 'sm') {
```

Comment:
so we only support xs and sm sizes?

Reply:
oh based on the story we can have other size. Maybe we need a default icon size, and overwrite it when it's `xs` and `sm` ?

Reply:
@plbabin Yup! That's what's going on :). We have default `iconSize` of `24`, allowing the user to supply their own size. BUT... if the (Button) `size` is `sm` or `xs`, we force an iconSize.

Reply:
oh right default size is coming from the default prop value

I will blame it on ☕️not being strong enough 🤣